Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java

```diff
@@ -534,4 +559,143 @@ private static boolean checkCompletedChain(Map<String, CertificateDTO> chain) {
+        this.organizationalUnit = organizationalUnit;
+    }
+
+    public String getOrganization() {
+        return organization;
+    }
+
+    public void setOrganization(String organization) {
+        this.organization = organization;
+    }
+
+    public String getCountryName() {
+        return countryName;
+    }
+
+    public void setCountryName(String countryName) {
+        this.countryName = countryName;
+    }
+
+    public int getImportWizardTabIndex() {
+        return importWizardTabIndex;
+    }
+
+    public void setImportWizardTabIndex(int importWizardTabIndex) {
+        this.importWizardTabIndex = importWizardTabIndex;
+    }
+
+    public void createCertificate() {
+        if (StringUtils.isBlank(getAlias()) || StringUtils.isBlank(getReferenceNumber())
+            || StringUtils.isBlank(getOrganizationalUnit()) || StringUtils.isBlank(getOrganization())
+            || StringUtils.isBlank(getCountryName())) {
+            HelperUtil.addMessageError(null, "Required fields cannot be empty when creating certificate.");
+        }
+
+        boolean status = service.createCertificate(getAlias(), getReferenceNumber(), getOrganizationalUnit(),
+            getOrganization(), getCountryName());
+        if (status) {
+            HelperUtil.addMessageInfo(null,
+                MessageFormat.format("Successfully created certifcate for: {0}", getAlias()));
+            importWizardTabIndex = TABINDEX_CREATECSR;
+        } else {
+            HelperUtil.addMessageError(null, MessageFormat.format("Failed to create certifcate for: {0}", getAlias()));
+        }
+    }
+
+    public void createCSR() {
+        if (StringUtils.isBlank(getAlias())) {
+            HelperUtil.addMessageError(null, "Required fields cannot be empty when creating CSR.");
+            return;
+        }
+        csrFile = null;
+
+        SimpleCertificateResponseMessageType response = service.createCSR(getAlias());
+
+        if (null != response && response.isStatus()) {
+            HelperUtil.addMessageInfo(null,
+                MessageFormat.format("Successfully created CSR for: {0}", getAlias()));
+            if(null != response.getCsrData()){
+                DataHandler data = response.getCsrData();
+                String filename = MessageFormat.format("{0}-{1}.csr", getAlias(), CoreHelpUtils.formatDate("yyyyMMdd", new Date()));
+                try {
+                    csrFile = new DefaultStreamedContent(data.getInputStream(), data.getContentType(), filename);
+                } catch (IOException e) {
+                    LOG.error("unable to open input-stream from response", e.getLocalizedMessage(), e);
+                }
+            }
+        } else {
+            HelperUtil.addMessageError(null, MessageFormat.format("Failed to create CSR for: {0}", getAlias()));
+        }
+    }
+
+    public StreamedContent getCsrFile() {
+        return csrFile;
+    }
+
+    public void cancelGatewayNew() {
```

Comment:
I still find this naming convention confusing. We are deleting the temporary keystore file, so why not deleteTemporaryKeystore?

Reply:
this mapped to cancel-button

Reply:
That's fine. But you have "cancel" and "new" in the same method. Its the same issue with "DeleteGatewayNew"